util/src/main/java/org/datacommons/util/CoordinatesResolver.java

```diff
@@ -0,0 +1,96 @@
+package org.datacommons.util;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import org.datacommons.proto.Mcf.McfGraph.PropertyValues;
+import org.datacommons.proto.Mcf.McfGraph.Values;
+import org.datacommons.proto.Mcf.ValueType;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest.Coordinate;
+import org.datacommons.proto.Recon.ResolveCoordinatesResponse;
+import org.datacommons.proto.Recon.ResolveCoordinatesResponse.Place;
+
+/** Resolves nodes with lat-lngs by calling the DC coordinates resolution API. */
+final class CoordinatesResolver {
+  private final Set<Coordinate> resolveCoordinates = ConcurrentHashMap.newKeySet();
+
+  private final ConcurrentHashMap<Coordinate, Set<String>> resolvedCoordinates =
+      new ConcurrentHashMap<>();
+
+  private final ReconClient client;
+
+  CoordinatesResolver(ReconClient client) {
+    this.client = client;
+  }
+
+  boolean submit(PropertyValues node) {
+    return getCoordinate(node).map(resolveCoordinates::add).map(unused -> true).orElse(false);
```

Comment:
nit: This looks like a slightly convoluted syntax to add to a set?

Reply:
Updated. I typically don't like to use get() on Optional but it makes sense to do here for readability. Thanks for calling it out.